components/MessageSelector.tsx

```diff
@@ -0,0 +1,44 @@
+import { MessageAction } from '../util/messagehelpers'
+import { ChangeEvent } from 'react'
+
+export function MessageSelector({ actions }: { actions: MessageAction[] }) {
+  const items = actions.map((action, i: number) => {
+    let item
+    if (action.enabled()) {
+      item = (
+        <a href={action.href} onClick={action.action}>
+          {action.label}
+        </a>
+      )
+    } else {
+      item = <a href="#">{action.label}</a>
+    }
+    return (
+      <option disabled={!action.enabled()} value={i} key={i}>
+        {action.label}
+      </option>
+    )
+  })
+
+  const handleItemSelected = (e: ChangeEvent<HTMLSelectElement>) => {
+    if (!e.target) {
+      return
+    }
+    const i = e.target.selectedIndex - 1
+    const action = i >= 0 && actions?.length > i ? actions[i] : undefined
```

Comment:
Since undefined is the TypeScript/JS default for a value that isn't present, I prefer to use it in that case. I like to use null for "the user explicitly set the value to 'not a value'"